api/data_explorer/controllers/export_url_controller.py

```diff
@@ -132,11 +142,98 @@ def _create_signed_url(gcs_path):
     return signed_url
 
 
+def _get_clause(column, type, value):
+    if type == 'text':
+        clause = column + ' = "' + value + '"'
+    elif type == 'boolean':
+        clause = column + ' =' + value
+    else:
+        clause = _get_range_clause(column, value)
+    return clause
+
+
+def _get_range_clause(column, value):
+    arr = value.split('-')
+    if len(arr) > 1:
+        low = arr[0]
+        high = arr[1]
+    else:
+        return column + " = " + value
+    if low.endswith('M'):
+        low = int(low[:-1])
+        high = int(high[:-1])
+        low = low * 1000000
+        high = high * 1000000
+        # For the million and billion ranges, there is overlap between the ranges.
+        # So we don't include the upper end of the range.
+        return column + " >= " + str(low) + " AND " + column + " < " + str(high)
+    elif low.endswith('B'):
+        low = int(low[:-1])
+        high = int(high[:-1])
+        low = low * 1000000000
+        high = high * 1000000000
+        return column + " >= " + str(low) + " AND " + column + " < " + str(high)
+
+    # For the ordinary ranges, there is no overlap between the ranges.
+    # So we include the upper end of the range. 
+    return column + " >= " + str(low) + " AND " + column + " <= " + str(high)
+
+
+def _get_filter_query(filter):
+    if filter is None or len(filter) == 0:
+        return ""
+    facets = current_app.config['UI_FACETS']
+    table_clauses = dict()
+    for f in filter:
```

Comment:
FYI - After Bryan's stuff is in, we'll have to exclude files facets from the SQL query somehow. I'll talk to Brad about this.